src/loaders/common/loader_helper.py

```diff
@@ -170,6 +169,20 @@ def find_free_port():
         return s.getsockname()[1]
 
 
+def create_fatal_dict_doc(error_message, source_file_path, stacktrace=None):
```

Comment:
Hi, May I get more insight on this one? I basically formatted the `FATAL_ERROR_FILES` structure as you suggested in [RE2022-196](https://kbase-jira.atlassian.net/jira/software/projects/RE2022/boards/78?selectedIssue=RE2022-196). Write out a merged `FATAL_ERROR_FILE` that sweep all the tools' batch directories looking for `FATAL_ERROR_FILES`

Reply:
So the logic and file format you're writing is fine. What I'm concerned about is that the tool files are being written in multiple different places, and therefore, in the future, someone could change things in one place but not in the others. As such, I want a single function that the tools pass their error data to and that function writes out the file, so the entire data -> file flow is handled in one spot.

For the parser error file it's less important since that's only called in one place, but since the formats are very similar it makes sense for both functions to work the same way and live in the same place.

E.g. https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
Maybe a list of NamedTuples with the 4 fields is enough? (kbase_id, error_message, source_file_path, stacktrace)

Reply:
sounds good

Reply:
Updated!

Reply:
:+1: